docs/docs/contributing.md

```diff
@@ -0,0 +1,73 @@
+---
+slug: /contributing/
+sidebar_position: 7
+---
+
+# Contributing
+
+Contributions to Task are very welcome, but we ask that you read this document before submitting a PR.
+
+## Before you start
+
+- **Check existing work** - Is there an existing PR? Are there issues discussing the feature/change you want to make? Please make sure you consider/address these discussions in your work.
+- **Backwards compatibility** - Will your change break existing Taskfiles? It is much more likely that your change will merged if it backwards compatible. Is there an approach you can take that maintains this compatibility? If not, consider opening an issue first so that API changes can be discussed before you invest you time into a PR.
+
+## 1. Setup
+
+- **Golang** - Task is written in [Go]. You will need to [install version 1.18+].
+- **Node.js** - [Node.js] is used to host Task's documentation server and is required if you want to run this server locally.
+- **Yarn** - [Yarn] is the Node.js package manager used by Task.
+
+## 2. Making changes
+
+- **Code style** - Try to maintain the existing code style where possible and ensure that code is formatted by `gofmt`.
```

Comment:
We also use `golangci-lint`. There's a `lint` command on the Taskfile to run it, and CI will fail if it found any offenses as well.